java/src/test/java/com/google/appengine/tools/pipeline/PipelineTest.java

```diff
@@ -15,54 +15,49 @@
 package com.google.appengine.tools.pipeline;
 
 import static com.google.appengine.tools.pipeline.impl.util.GUIDGenerator.USE_SIMPLE_GUIDS_FOR_DEBUGGING;
+import static org.mockito.Mockito.mock;
 
 import com.google.appengine.api.taskqueue.dev.LocalTaskQueue;
-import com.google.appengine.api.taskqueue.dev.QueueStateInfo;
-import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
 import com.google.appengine.tools.development.testing.LocalModulesServiceTestConfig;
 import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
 import com.google.appengine.tools.development.testing.LocalTaskQueueTestConfig;
+import com.google.appengine.tools.pipeline.impl.PipelineManager;
+import com.google.appengine.tools.pipeline.impl.backend.AppEngineBackEnd;
+import com.google.appengine.tools.pipeline.impl.backend.SerializationStrategy;
 import com.google.apphosting.api.ApiProxy;
 
-import junit.framework.TestCase;
+import com.google.auth.Credentials;
 import lombok.Getter;
-
-import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 
 /**
  * @author rudominer@google.com (Mitch Rudominer)
  *
  */
-public abstract class PipelineTest extends TestCase {
+@PipelineSetupExtensions
+public abstract class PipelineTest {
 
   protected LocalServiceTestHelper helper;
   protected ApiProxy.Environment apiProxyEnvironment;
 
   private static StringBuffer traceBuffer;
 
+  public static final String PROJECT = DatastoreExtension.TEST_DATASTORE_PROJECT_ID;
+
   @Getter
   private LocalTaskQueue taskQueue;
 
+  protected PipelineService pipelineService;
+  protected PipelineManager pipelineManager;
+  protected AppEngineBackEnd appEngineBackend;
```

Comment:
these are being used to "inject" these values to `test_`() methods in subclasses; future can do this via ParameterResolution, declaring them in each test method